devito/ops/node_factory.py

```diff
@@ -45,6 +45,10 @@ def new_ops_arg(self, indexed, is_write):
                 indexed.dtype,
                 not is_write
             )
+
+            symbol_to_access._origin_name = indexed.function.name
```

Comment:
this is wrong. You should never ever change state. The only thing you can do is to supply the arguments at construction time via ``__init__``

What you're doing here breaks immutability

Reply:
Fixed.